docs/params.rst

```diff
@@ -0,0 +1,94 @@
+==========
+Parameters
+==========
+
+:ref:`Spider arguments <spiderargs>` have some limitations:
+
+-   There is no standard way for spiders to indicate which parameters they
+    expect or support.
+
+-   Since arguments can come from the command line, and those are always
+    strings, you have to write code to convert arguments to a different type
+    when needed.
+
+-   If you want argument validation, such as making sure that arguments are of
+    the right type, or that required arguments are present, you must implement
+    validation yourself.
+
+scrapy-spider-metadata allows overcoming those limitations.
+
+.. _define-params:
+
+Defining supported parameters
+=============================
+
+To define spider parameters, define a subclass of `pydantic.BaseModel`_,
+and then make your spider also inherit from
+:class:`~scrapy_spider_metadata.Args` with your parameter specification class
+as its parameter:
+
+.. _pydantic.BaseModel: https://docs.pydantic.dev/latest/api/base_model/#pydantic.BaseModel
+
+.. code-block:: python
+
+    from pydantic import BaseModel
+    from scrapy import Request, Spider
+    from scrapy_spider_metadata import Args
+
+    class MyParams(BaseModel):
+        pages: int
+
+    class MySpider(Args[MyParams], Spider):
+        name = "my_spider"
+
+        def start_requests(self):
+            for index in range(1, self.args.pages + 1):
+                yield Request(f"https://books.toscrape.com/catalogue/page-{index}.html")
+
+
+To learn how to define parameters in your `pydantic.BaseModel`_
+subclass, see the `Pydantic usage documentation
+<https://docs.pydantic.dev/latest/usage/models/>`_.
+
+Defined parameters make your spider:
+
+-   Halt with an exception if there are missing arguments or any provided
+    argument does not match the defined parameter validation rules.
```

Comment:
What's interesiting is that it seems if there are any extra arguments, they're going to be skipped by default - see https://docs.pydantic.dev/latest/concepts/models/#extra-fields. This behavior looks fine to me.